R/adl.R

```diff
@@ -122,4 +121,160 @@ adl_fun <- function (ADL_01, ADL_02, ADL_03, ADL_04, ADL_05) {
+#'
+#' library(cchsflow)
+#' adl2001 <- rec_with_table(
+#'   cchs2001_p, c(
+#'     "ADL_01", "ADL_02", "ADL_03", "ADL_04", "ADL_05", "ADL_score_5"
+#'   )
+#' )
+#'
+#' head(adl2001)
+#'
+#' adl2009_2010 <- rec_with_table(
+#'   cchs2009_2010_p, c(
+#'     "ADL_01", "ADL_02", "ADL_03", "ADL_04", "ADL_05", "ADL_score_5"
+#'   )
+#' )
+#'
+#' tail(adl2009_2010)
+#'
+#' combined_adl <- bind_rows(adl2001, adl2009_2010)
+#'
+#' head(combined_adl)
+#'
+#' tail(combined_adl)
+#'
+#' # Using adl_score_5_fun() to generate to ADL_score_5 based on user inputted values.
+#' #
+#' # Let's say you do not need help preparing meals, you need help getting to
+#' # appointments or errands, you need help doing housework, do not need help
+#' # doing personal care, and do not need help moving inside the house. Using
+#' # adl_score_5_fun() we can check the number of tasks you need help with
+#'
+#' ADL_score_5 <- adl_score_5_fun(2, 1, 1, 2, 2)
+#'
+#' print(ADL_score_5)
+#'
+adl_score_5_fun <-
+  function(ADL_01, ADL_02, ADL_03, ADL_04, ADL_05) {
+    # For each row in the ADL variables, count the total number of "missing"
+    # values in each row
+    count_missing_adl <- rep(0, length(ADL_01))
+    # For each row of each of the above ADL variables, if they have a value of 
+    # "missing" then increment the count for that row in the above
+    # vector by one
+    count_missing_adl <-
+      ifelse(ADL_01 == "NA(b)", count_missing_adl + 1, count_missing_adl)
+    count_missing_adl <-
+      ifelse(ADL_02 == "NA(b)", count_missing_adl + 1, count_missing_adl)
+    count_missing_adl <-
+      ifelse(ADL_03 == "NA(b)", count_missing_adl + 1, count_missing_adl)
+    count_missing_adl <-
+      ifelse(ADL_04 == "NA(b)", count_missing_adl + 1, count_missing_adl)
+    count_missing_adl <-
+      ifelse(ADL_05 == "NA(b)", count_missing_adl + 1, count_missing_adl)
+    
+    # For each row in the ADL variables, count the total number of "not
+    # applicable" values in each row
+    count_not_applicable_adl <- rep(0, length(ADL_01))
+    # For each row of each of the above ADL variables, if they have a value of 
+    # "not applicable" then increment the count for that row in the above
+    # vector by one
+    count_not_applicable_adl <-
+      ifelse(ADL_01 == "NA(a)",
+             count_not_applicable_adl + 1,
+             count_not_applicable_adl)
+    count_not_applicable_adl <-
+      ifelse(ADL_02 == "NA(a)",
+             count_not_applicable_adl + 1,
+             count_not_applicable_adl)
+    count_not_applicable_adl <-
+      ifelse(ADL_03 == "NA(a)",
+             count_not_applicable_adl + 1,
+             count_not_applicable_adl)
+    count_not_applicable_adl <-
+      ifelse(ADL_04 == "NA(a)",
+             count_not_applicable_adl + 1,
+             count_not_applicable_adl)
+    count_not_applicable_adl <-
+      ifelse(ADL_05 == "NA(a)",
+             count_not_applicable_adl + 1,
+             count_not_applicable_adl)
```

Comment:
This is too similar to your NA(b) code there should be a way to collapse it together